lib/ddtrace/contrib/extensions.rb

```diff
@@ -1,21 +1,34 @@
 require 'set'
-require 'ddtrace/contrib/registry'
+require 'datadog/contrib'
+require 'ddtrace/configuration/settings'
 
 module Datadog
   module Contrib
     # Extensions that can be added to the base library
     # Adds registry, configuration access for integrations.
+    #
+    # DEV: The Registry should probably be part of the core tracer
+    # as it represents a global tracer repository that is strongly intertwined
+    # with the tracer lifecycle and deeply modifies the tracer initialization
+    # process.
+    # Most of this file should probably live inside the tracer core.
     module Extensions
       def self.extended(base)
-        Datadog.extend(Helpers)
-        Datadog.extend(Configuration)
-        Datadog::Configuration::Settings.include(Configuration::Settings)
+        base.extend(Helpers)
+        base.extend(Configuration)
       end
 
       # Helper methods for Datadog module.
       module Helpers
+        # Returns the global integration registry.
+        #
+        # This method is not safe to use while the tracer is initializing,
+        # thus access to the registry should go through
+        # ::Datadog::Contrib::REGISTRY for internal tracer work.
```

Comment:
Minor: Should we also document this near the constant? Just in case someone misses this comment.